### PR TITLE
[CLOUD-3395] - Update legacy modules for EAP CD18 build

### PR DESCRIPTION
**Pull request by luck3y**
Signed-off-by: Ken Wills <kwills@redhat.com>
https://issues.jboss.org/browse/CLOUD-3395

### CODE REVIEW

**Comment by luck3y** [comment]
FYI this includes a ported standalone-openshift.xml created from CD18 and comparing with CD17, here: https://github.com/jboss-container-images/jboss-eap-modules/blob/7.3.x-legacy/jboss-eap-config-openshift/CD17.0/added/standalone-openshift.xml and with the standalone-full-ha.xml in the CD18 distro.

We don't intend to build an image with this, this configuration is intended for layered-products to use.

**Comment by luck3y** [comment]
Requires: https://github.com/jboss-container-images/jboss-eap-7-image/pull/165

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+                <cache name="distributable" passivation-store-ref="infinispan" aliases="passivating clustered"/>
+            </caches>
+            <passivation-stores>
+                <passivation-store name="infinispan" cache-container="ejb" max-size="10000"/>
+            </passivation-stores>
+            <async thread-pool-name="default"/>
+            <!-- ##TIMER_SERVICE## -->
+            <remote connector-ref="http-remoting-connector" thread-pool-name="default">
+                <channel-creation-options>
+                    <option name="MAX_OUTBOUND_MESSAGES" value="1234" type="remoting"/>
```
Comment (jfdenise) [inline jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: \<option name="READ_TIMEOUT" value="${prop.remoting-connector.read.timeout:20}" type="xnio"/\>
is present in the generated galleon configuration.

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+                <smtp-server outbound-socket-binding-ref="mail-smtp"/>
+            </mail-session>
+        </subsystem>
+        <subsystem xmlns="urn:jboss:domain:messaging-activemq:8.0">
+            <!-- ##MESSAGING_SUBSYSTEM_CONFIG## -->
+        </subsystem>
+        <subsystem xmlns="urn:wildfly:microprofile-config-smallrye:1.0">
+            <!-- ##MICROPROFILE_CONFIG_SOURCE## -->
+        </subsystem>
+        <subsystem xmlns="urn:wildfly:microprofile-health-smallrye:2.0" security-enabled="false" empty-liveness-checks-status="${env.MP_HEALTH_EMPTY_LIVENESS_CHECKS_STATUS:UP}" empty-readiness-checks-status="${env.MP_HEALTH_EMPTY_READINESS_CHECKS_STATUS:UP}"/>
```
Comment (jfdenise) [inline jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: security-enabled="false" is not present in the galleon generated config and was not present in CD17.

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml (hunk 1 above)
Comment (bstansberry) [reply jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: @jfdenise That is a bug in the galleon generated config

https://github.com/wildfly/wildfly/commit/6894e74c09de9221e7c6966cf90f970e3fffd3e9#diff-e6ce645a4e9818d119c2ec6bad8d4c73

It is not a serious problem for CD 18 but is a minor thing to clean up.

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml (hunk 2 above)
Comment (bstansberry) [reply jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: @jfdenise That seems like a flaw in the galleon generated config as security-enabled="false" is in WF 18 standalone-full-ha.xml and is in the feature-group spec:

https://github.com/wildfly/wildfly/blob/master/galleon-pack/src/main/resources/feature_groups/metrics.xml#L4

Comment (bstansberry) [reply jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: Oops; this is about health and I linked metrics. But same thing there:

https://github.com/wildfly/wildfly/blob/master/galleon-pack/src/main/resources/feature_groups/health.xml#L4

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml (hunk 1 above)
Comment (jfdenise) [reply jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: @bstansberry , ok. The galleon config explicitly add it back in order to align with CD17. I will fix it.

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml (hunk 2 above)
Comment (jfdenise) [reply jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: This was explicitly removed to align with CD17, will add it back.
I logged CLOUD-3425 and CLOUD-3426

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml (hunk 1 above)
Comment (bstansberry) [reply jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: Thanks; that was a bug in CD 17 then; we should have detected that change in the WF file from 16 to 17 and made the same change in the CD17 file.

File: jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+        <subsystem xmlns="urn:jboss:domain:batch-jberet:2.0">
+            <!-- ##DEFAULT_JOB_REPOSITORY## -->
+            <default-thread-pool name="batch"/>
+            <job-repository name="in-memory">
+                <in-memory/>
+            </job-repository>
+            <!-- ##JOB_REPOSITORY## -->
+            <thread-pool name="batch">
+                <max-threads count="10"/>
+                <keepalive-time time="30" unit="seconds"/>
```
[the remark this answers is not in the document]
Comment (luck3y) [reply jboss-eap-config-openshift/CD18.0/added/standalone-openshift.xml]: Updated, this should be all set now.

**Comment by luck3y** [comment]
This should be all set now if someone could take a final look.

**Pull request closed without merging**